lib/SILOptimizer/Differentiation/Common.cpp

```diff
@@ -257,6 +257,8 @@ findDebugLocationAndVariable(SILValue originalValue) {
   for (auto *use : originalValue->getUses()) {
     if (auto *dvi = dyn_cast<DebugValueInst>(use->getUser()))
       return dvi->getVarInfo().map([&](SILDebugVariable var) {
+        if (var.DIExpr.startsWithDeref())
+          var.DIExpr.eraseElement(var.DIExpr.element_begin());
```

Comment:
I think this would benefit from a comment, since it's unexpected that a function named findXXX() is modifying var.